src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs

```diff
@@ -85,13 +86,40 @@ public string GetRepoFetchUrl(string org, string repository)
             return $"https://{_parameters.GitHubUser}:{_parameters.GitHubToken}@github.com/{org}/{repository}";
         }
 
+        public string GetAzDoRepoUrl(string repoName, string azdoAccount = "", string azdoProject = "")
+        {
+            if (String.IsNullOrEmpty(azdoAccount))
+            {
+                azdoAccount = "dnceng";
+            }
+
+            if (string.IsNullOrEmpty(azdoProject))
+            {
+                azdoProject = "internal";
+            }
+
+            return $"https://dev.azure.com/{azdoAccount}/{azdoProject}/_git/{repoName}";
+        }
+
+        public Task<string> RunDarcAsyncWithInput(string input, params string[] args)
+        {
+            return TestHelpers.RunExecutableAsyncWithInput(_parameters.DarcExePath, input, args.Concat(new[]
```

Comment:
It's... interesting that we are importing the Darc project... we could just... call the operations we want rather than having to build up the giant exe command line.

Reply:
That would also mean we didn't have to do all the string parsing on the output of the process. :-)

Reply:
I really like that idea, but would it reduce the code coverage we want? Right now I'm explicitly only using it to build expected results, not to do any of the test calls.

Reply:
Makes sense. Let get it in as is for now, and we can think about changes later (that probably goes for a lot of my comments)